js/interface.js

```diff
@@ -307,32 +311,113 @@
     return Fliplet.API.request({
       url: [
         'v1/widgets?include_instances=true&tags=type:menu',
+        '&include_all_versions=true',
         '&appId=' + Fliplet.Env.get('appId'),
-        '&organizationId=' + (Fliplet.Env.get('organizationId') || '')
+        '&organizationId=' + (organizationId || '')
       ].join('')
     }).then(function(response) {
       return Promise.resolve(response.widgets);
     });
   }
 
+  /**
+   * Function to filter organizationMenus by version.
+   * @param {Array} data - array of organizationMenus
+   * @param {Object} latestVersionMenu - current active menu
+   * @returns {Object} latestVersionMenu - object which contains the latest version of current menu package type
+   */
+
+  function getLatestMenuVersion(data, latestVersionMenu) {
+    var previousVersion;
+
+    data.forEach(function(menu) {
+      var formattedMenuVersion = menu.version.split('.').join('');
+      var formattedPreviousMenuVersion = previousVersion ? previousVersion.version.split('.').join('') : null;
+
+      // Compare each record with the previous one by version to find the newest version.
+      if (previousVersion && formattedPreviousMenuVersion !== formattedMenuVersion) {
+        latestVersionMenu = (formattedPreviousMenuVersion > formattedMenuVersion) ? previousVersion : null;
+      }
+
+      previousVersion = menu;
+
+      // Always write the system menu first, then if iteration finds the newest version, latestVersionMenu will be overwritten.
+      if (!latestVersionMenu) {
+        latestVersionMenu = menu;
+      }
+    });
+
+    return latestVersionMenu;
+  }
+
+  /**
+   * Creates a list of menu widgets to be displayed based on versions and current usage
+   * @param {Array} menus - List of menu widgets
+   * @returns {Array} List of menu widgets to be displayed based on versions and current usage
+   */
+
+  function generateMenuList(menus) {
+    var result = [];
+    var hasActiveMenu = false;
+
+    // Cycle iterating the array of system menus to fulfill the necessary conditions.
+    menus.forEach(function(item) {
+      var latestVersionMenu;
+
+      // Creating an array of one type of menu.
+      var packageTypeMenus = menus.filter(function(menu) {
```

Comment:
I suggest you use something like `_.groupBy(menus, 'package')` to group the menus and loop through them.

The way you're using `.filter()` inside a `.forEach()` is not as efficient.